bindings/matrix-sdk-ffi/src/platform.rs

```diff
@@ -245,12 +231,65 @@ pub struct TracingFileConfiguration {
     max_files: Option<u64>,
 }
 
+#[derive(PartialEq, PartialOrd)]
+enum LogTarget {
```

Comment:
I think EXA has more target than that https://github.com/element-hq/element-x-android/blob/65ce91a8fbe287564bbdaa969ce9b925edacdb6c/libraries/matrix/api/src/main/kotlin/io/element/android/libraries/matrix/api/tracing/TracingFilterConfiguration.kt#L48-L66

Would be nice to have a `CustomTarget(rawString)`? that would allow to add a new target wihtout having to compile a custom sdk when developping?

Reply:
> I think EXA has more target than that

Seems so. The missing ones are `matrix_sdk_ffi::uniffi_api` , `matrix_sdk` and `matrix_sdk::send_queue`. Not sure if we should include them or not to be honest ¯\_(ツ)_/¯ Perhaps keep `matrix_sdk` on info and not add the rest?


> Would be nice to have a CustomTarget(rawString)? 

Thought about exposing an optional `filter` parameter that will just override everything but I figured it's best we keep the api simple as a first step. I'm not sure how often you'll want to debug something with custom tracing on the app side without having a custom sdk version as well, moment in which tweaking the logs becomes easy.

Reply:
> Thought about exposing an optional filter parameter that will just override everything but I figured it's best we keep the api simple as a first step. I'm not sure how often you'll want to debug something with custom tracing on the app side without having a custom sdk version as well, moment in which tweaking the logs becomes easy.

Could the `TracingConfiguration` include an `extra_logs: Vec<(String, LogLevel)>`, that would be applied as such without any checks? Having some dynamic way to set some logs is definitely a recurring need, and will pay off super quick.

Then we wouldn't need the mandatory `app_target` field in there, as it could be part of the extra logs. (And that gives you the possibility to use different spans if you'd like!)

Reply:
Oh yeah, nice idea, let's do it! 💪

Reply:
Okay, so I had a play and ended up with passing in *only* an array of `target`s as opposed to `(target, level)` as doing the latter would force the final client to duplicate the level adjustment logic done here. 

If we pass in only the targets we can keep using the same logic while offering just enough flexibility imho.